eval/crypt.go

```diff
@@ -153,7 +153,7 @@ func DecryptSecrets(ctx context.Context, filename string, source []byte, decrypt
 
 		// Don't attempt to decrypt secrets outside of "values"
 		path := n.Syntax().Path()
-		if !strings.HasPrefix(path, "values.") {
+		if !strings.HasPrefix(path, "values.") && !strings.HasPrefix(path, "values[") {
```

Comment:
@pgavlin this works, but do we have a nicer way of checking if something is within the toplevel `values`?

Reply:
We could inspect the structure of the document in `rewriteYAML` and only rewrite the value of the `values` key, if any exists. Something like

```go
if obj, ok := syn.(*syntax.ObjectNode); ok {
	for i := 0; i < obj.Len(); i++ {
		kvp := obj.Index(i)
		if kvp.Key.Value() == "values" {
			v, diags, err := syntax.Walk(kvp.Value)
			if err != nil {
				return nil, err
			}
			if len(diags) != 0 {
				return nil, diags
			}
			kvp.Value = v
			obj.SetIndex(i, kvp)
			break
		}
	}
}
```

Reply:
the issue with that is we also use `rewriteYAML` on encrypt and i want that to rewrite the whole doc (in case someone accidentally puts a secret in an invalid top-level) - which maintains current behavior

I've updated this PR to use a more elegant check using property paths